hoodie-client/src/main/java/com/uber/hoodie/table/HoodieMergeOnReadTable.java

```diff
@@ -203,14 +207,29 @@ public HoodieCompactionPlan scheduleCompaction(JavaSparkContext jsc, String inst
             return false;
           };
 
+          final Map<FileStatus, Boolean> filesToDeletedStatus = new HashMap<>();
+
           switch (instant.getAction()) {
             case HoodieTimeline.COMMIT_ACTION:
+              try {
+                // Rollback of a commit should delete the newly created parquet files along with any log
+                // files created with this as baseCommit. This is required to support multi-rollbacks in a MOR table.
+                super.deleteCleanedFiles(filesToDeletedStatus, partitionPath, filter);
+                hoodieRollbackStats = HoodieRollbackStat.newBuilder()
+                    .withPartitionPath(partitionPath).withDeletedFileResults(filesToDeletedStatus).build();
+                break;
+              } catch (IOException io) {
+                throw new UncheckedIOException("Failed to rollback for commit " + commit, io);
+              }
             case HoodieTimeline.COMPACTION_ACTION:
               try {
-                Map<FileStatus, Boolean> results = super
-                    .deleteCleanedFiles(partitionPath, Collections.singletonList(commit));
+                // Rollback of a compaction action means that the compaction is scheduled and has not yet finished.
+                // In this scenario we should delete only the newly created parquet files and not corresponding base
+                // commit log files created with this as baseCommit since updates would have been written to the log
+                // files.
+                super.deleteCleanedFiles(filesToDeletedStatus, commits, partitionPath);
```

Comment:
We also need to cleanup log files because of delta commits after pending compaction. Right? In that sense, both the cases : case HoodieTimeline.COMMIT_ACTION and case HoodieTimeline.COMPACTION_ACTION must be handled in a same way. right ?